zebra-rpc/src/methods/get_block_template_rpcs.rs

```diff
@@ -250,35 +261,20 @@ where
         // Since this is a very large RPC, we use separate functions for each group of fields.
         async move {
             let _tip_height = best_chain_tip_height(&latest_chain_tip)?;
+            let mempool_txs = select_mempool_transactions(mempool).await?;
 
-            // TODO: put this in a separate get_mempool_transactions() function
-            let request = mempool::Request::FullTransactions;
-            let response = mempool.oneshot(request).await.map_err(|error| Error {
-                code: ErrorCode::ServerError(0),
-                message: error.to_string(),
-                data: None,
-            })?;
-
-            let transactions = if let mempool::Response::FullTransactions(transactions) = response {
-                // TODO: select transactions according to ZIP-317 (#5473)
-                transactions
-            } else {
-                unreachable!("unmatched response to a mempool::FullTransactions request");
-            };
+            let miner_fee = miner_fee(&mempool_txs);
 
-            let merkle_root;
-            let auth_data_root;
+            /*
+            TODO: create a method Transaction::new_v5_coinbase(network, tip_height, miner_fee);
```

Comment:
This might need to be part of the same call as the `FullTransactions` request to guarantee nothing else is added or evicted between these calls.

Reply:
If that happened, what problems could it cause?

Either way, this seems like something we can deal with in the next PR.